core/src/main/java/com/scalar/db/transaction/consensuscommit/TwoPhaseConsensusCommitManager.java

```diff
@@ -90,7 +90,11 @@ public TwoPhaseConsensusCommitManager(DatabaseConfig databaseConfig) {
     coordinator = new Coordinator(storage, config);
     parallelExecutor = new ParallelExecutor(config);
     recovery = new RecoveryHandler(storage, coordinator, tableMetadataManager);
-    commit = new CommitHandler(storage, coordinator, tableMetadataManager, parallelExecutor);
+    // Create CommitHandler with throwExceptionIfCommittedTransactionExists false different from the
+    // case for ConsensusCommitManager because each service tries to put the state 'committed' in
+    // the transaction with a two-phase commit interface, and we want to ignore the existing
+    // committed state record.
```

Comment:
The last sentence sounds a bit weird to me.
How about something like this?

We want to avoid cases where some services get exceptions due to conflicting commits with the same transaction ID.

Reply:
Great, thank you. Revised, including the above comment, in b433e335d.